tools/autograd/templates/Functions.cpp

```diff
@@ -302,6 +302,19 @@ Tensor cumsum_backward(const Tensor & x, int64_t dim) {
 Tensor unsqueeze_to(const Tensor & self, IntList sizes) {
   auto result = self;
   int64_t nDims = sizes.size();
+
+  // Let's say the input had size (1, 1). input.squeeze(), with scalars
+  // disabled, produces a result of size (1,). This needs some
+  // special handling because for all other cases we unsqueeze every
+  // dimension that has size 1; doing that here would lead to one extra
+  // unsqueezed dimension
+  if (self.sizes().equals({1})) {
+    for (int64_t dim = 0; dim < nDims - 1; dim++) {
+      result = result.unsqueeze(dim);
```

Comment:
It would be nicer to use `view` for this - every unsqueeze in here is an autograd op, so you end up doing 5x the work (and the graph gets larger).